config/genesis/genesis.go

```diff
@@ -17,17 +17,17 @@ type Genesis struct {
 	Genesis    GenesisFields
 }
 
-// Genesis stores the data parsed from the genesis configuration file
+// GenesisData defines the genesis file data formatted for trie storage
 type GenesisData struct {
-	Name          string
-	Id            string
-	Bootnodes     [][]byte
-	ProtocolId    string
-	genesisFields GenesisFields
+	Name       string
+	Id         string
+	Bootnodes  [][]byte
+	ProtocolId string
 }
 
+// GenesisFields stores genesis raw data
 type GenesisFields struct {
-	Raw map[string]string
+	Raw [2]map[string]string
```

Comment:
is this necessarily 2, or can it just be a slice?

Reply:
I originally had a slice but I changed it to `2`. I am not quite sure how it will be used but it looks like the two mappings break down to`top` and `children` in substrate ([see substrate code](https://github.com/paritytech/substrate/blob/b443ddad65f2ffed68db4973b7b78231c6fc3674/client/chain-spec/src/chain_spec.rs#L77)).

Reply:
okay cool, makes sense to me